### DISCUSSION TITLE
Leader key not working the first time after starting nvim

### DISCUSSION THREAD

**Discussion by sgon00** [NvChad/NvChad#3149] [Q&A] [answered]
Hi, I am new to neovim. My leader key (space) always fails to work the first time I press it, regardless of how long I wait after starting nvim. On the first press of the leader key, I see <20> in the status. It works correctly from the second press onward.

Additionally, I occasionally encounter situations where it doesn’t work randomly (once in a while), but since I can’t reliably reproduce it and can't find the pattern so far, this question is just about the issue with the first press of the space key.

Is this normal behavior?

Please check the following for my environment:

My neovim version:

```
NVIM v0.10.3
Build type: Release
LuaJIT 2.1.1713484068
```

I am running Xubuntu 24.04 and xfce4-terminal.

**Comment by siduck** [comment]
can you show a video with keypress?  use showkeys plugin or some screenkey

**Reply by sgon00** [reply]
Thanks a lot for your reply. Please check the following two screencasts. One gif file and one mp4 video.

![Peek_nvchad_leader_issue](https://github.com/user-attachments/assets/666f60bc-5a8e-432d-9f20-f4feea55213f)


https://github.com/user-attachments/assets/4045096d-7921-4566-8205-3426ad5fb8e9

**Comment by siduck** [comment]
does this happen on non nvchad configs?

**Reply by sgon00** [reply]
I am new to neovim and nvchad. I just started to use neovim + nvchad the day I asked this question. I have no ideas how to do non-nvchad. For me, nvchad is neovim. Will leader key even work in non nvchad environment? I know nothing about non-nvchad neovim. Hopefully, I don't need to uninstall nvchad to test on non-nvchad. Thanks. :)

**Reply by siduck** [reply]
@sgon00 
```bash
 alias tiny='NVIM_APPNAME="tiny" nvim'
```

thats how i use nvim_appname feature of nvim to use multiple configs.

So clone any config in ~/.config/tiny   and nvim will take care of it. the alias to make things easy!

**Reply by sgon00** [reply]
Thanks a lot for this tip. I tried `tiny` command. It can run non-nvchad neovim. But I don't know what to look at. When I press space leader key in Normal mode, it won't pop up anything. So that means there is no leader key command in neovim. So I don't really understand your previous question regarding `does this happen on non nvchad configs?`. My question is leader key not working at the first time in nvchad. But pure neovim does not have leader key, so it won't work at all.

**Reply by siduck** [reply]
try other nvim configs. like lazyvim, astro or some minimal where u set leader as space. 

and tiny is just an example, u can name it anything

**Reply by sgon00** [reply]
Dear @siduck, I tried `lazyvim`, it's very fast. The leader key does NOT have this problem and the leader window can pop up very fast.

**Reply by siduck** [reply, marked as answer]
@sgon00  the leader window is lazyloaded in nvchad , put this in your plugins config

https://github.com/NvChad/NvChad/pull/3036 i didnt accept this pr as it breaks lazyloading

```lua
 {
    "folke/which-key.nvim",
    event = "VeryLazy",
  }

**Reply by sgon00** [reply]
@siduck Thank you very much for your help. This code works very well. I marked this code as the answer.
**Edited:** I put this to `~/.config/nvim/lua/plugins/init.lua`. Hopefully, that is the right place to do it. Cheers.
